templates/default/docker.yaml.erb

```diff
@@ -0,0 +1,50 @@
+# Generated by Chef, local modifications will be overwritten
+
+init_config:
+  <% if @init_config %>
+  <% if @init_config.key?('docker_root') -%>
+  docker_root: <%= @init_config['docker_root'] %>
+  <% end -%>
+  <% if @init_config.key?('socket_timeout') -%>
+  socket_timeout: <%= @init_config['socket_timeout'] %>
+  <% end -%>
+  <% end -%>
+
+instances:
+  <% @instances.each do |i| -%>
```

Comment:
I think we do have many cases where the user is shown an example in the recipe. See here for an example of the http check: https://github.com/DataDog/chef-datadog/blob/master/recipes/http_check.rb#L3-L20

This seems to be a better interface for now, and soon enough we might be able to replace a large amount of identical template files with one standardized file that any monitor would use.
I am considering a breaking change for a 3.0.0 release where we would convert all existing templates and move them to this method. But that's longer term.